lib/internal/test_runner/tap_parser.js

```diff
@@ -149,22 +151,26 @@ class TapParser extends Transform {
   processChunk(chunk) {
     const str = this.#lastLine + chunk.toString('utf8');
     const lines = StringPrototypeSplit(str, '\n');
-    this.#lastLine = ArrayPrototypeSplice(lines, lines.length - 1, 1)[0];
+    this.#lastLine = ArrayPrototypePop(lines);
 
-    let chunkAsString = lines.join('\n');
+    let chunkAsString = ArrayPrototypeJoin(lines, '\n');
     // Special case where chunk is emitted by a child process
-    chunkAsString = RegExpPrototypeSymbolReplace(
-      /\[out\] /g,
+    chunkAsString = StringPrototypeReplaceAll(
       chunkAsString,
+      '[out] ',
       ''
     );
-    chunkAsString = RegExpPrototypeSymbolReplace(
-      /\[err\] /g,
+    chunkAsString = StringPrototypeReplaceAll(
       chunkAsString,
+      '[err] ',
       ''
     );
-    chunkAsString = RegExpPrototypeSymbolReplace(/\n$/, chunkAsString, '');
-    chunkAsString = RegExpPrototypeSymbolReplace(/EOF$/, chunkAsString, '');
+    if (StringPrototypeEndsWith(chunkAsString, '\n')) {
+      chunkAsString = StringPrototypeSlice(chunkAsString, 0, chunkAsString.length - 1);
+    }
+    if (StringPrototypeEndsWith(chunkAsString, 'EOF')) {
+      chunkAsString = StringPrototypeSlice(chunkAsString, 0, chunkAsString.length - 3);
```

Comment:
```suggestion
      chunkAsString = StringPrototypeSlice(chunkAsString, 0, -1);
    }
    if (StringPrototypeEndsWith(chunkAsString, 'EOF')) {
      chunkAsString = StringPrototypeSlice(chunkAsString, 0, -3);
```
Passing negative integers to the end index behaves the same way, slicing from right to left.